path_login.go

```diff
@@ -51,13 +51,33 @@ func pathLogin(b *azureAuthBackend) *framework.Path {
 			logical.AliasLookaheadOperation: &framework.PathOperation{
 				Callback: b.pathLogin,
 			},
+			logical.ResolveRoleOperation: &framework.PathOperation{
+				Callback: b.pathResolveRole,
+			},
 		},
 
 		HelpSynopsis:    pathLoginHelpSyn,
 		HelpDescription: pathLoginHelpDesc,
 	}
 }
 
+func (b *azureAuthBackend) pathResolveRole(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	roleName := data.Get("role").(string)
```

Comment:
Can do, though I was particularly wary of changing the plugin behaviour as part of this change (in both cases, I kept it the same as the login function). I could add a `getFieldValueStr` function here too but it could change plugin behaviour and I wouldn't want the role resolution function to be different to the login function.

Should add `getFieldValueStr` as in the Kubernetes plugin, knowing that it does -- subtly -- change the logic?

Reply:
Oh I see, fair enough. I'm relaxed about which way to go with this, I think it's fine as-is 👍